lib/components/ClinicUserSelect.js

```diff
@@ -0,0 +1,162 @@
+*
+* This program is distributed in the hope that it will be useful, but WITHOUT
+* ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+* FOR A PARTICULAR PURPOSE. See the License for more details.
+*
+* You should have received a copy of the License along with this program; if
+* not, you can obtain one from Tidepool Project at tidepool.org.
+* == BSD2 LICENSE ==
+*/
+
+var _ = require('lodash');
+var React = require('react');
+var Select = require('react-select');
+var sundial = require('sundial');
+var cx = require('classnames');
+
+var styles = require('../../styles/components/ClinicUserSelect.module.less');
+ 
+var ClinicUserSelect = React.createClass({
+  propTypes: {
+    allUsers: React.PropTypes.object.isRequired,
+    onUserChange: React.PropTypes.func.isRequired,
+    targetId: React.PropTypes.string,
+    targetUsersForUpload: React.PropTypes.array.isRequired,
+    onAddUserClick: React.PropTypes.func.isRequired,
+    setTargetUser: React.PropTypes.func.isRequired
+  },
+
+  handleChange: function(e) {
+    this.props.setTargetUser(e);
+  },
+
+  handleClickNext: function(e) {
+    e.preventDefault();
+    if(this.props.targetId){
+      this.props.onUserChange(this.props.targetId);
+    }
+  },
+
+  valueRenderer: function(option) {
+    var user = _.get(this.props.allUsers, option.value);
+    var name = _.get(user, 'fullName');
+    var bday = _.get(user, ['patient', 'birthday'], '');
+    var mrn = _.get(user, ['patient', 'mrn'], '');
+
+    var dateString = sundial.translateMask(bday, 'YYYY-MM-DD', 'M/D/YYYY');
+
+    return (
+      <div className={styles.optionLabelWrapper}>
+        <div className={styles.optionLabelName}>
+          {name} {mrn ? 'MRN:'+mrn : ''}
+        </div>
+        <div className={styles.optionLabelBirthday}>
+          {dateString}
+        </div>
+      </div>
+    );
+  },
+
+  renderError: function() {
+    var self = this;
+    if(this.props.updateProfileErrorMessage && !this.props.updateProfileErrorDismissed){
+      return (
+        <div className={styles.error}>
+          {this.props.updateProfileErrorMessage}
+          <i className={styles.iconClose} onClick={this.props.dismissUpdateProfileError}></i>
+        </div>
+      );
+    }
+  },
+
+  renderSelector: function(){
+    var allUsers = this.props.allUsers;
+    var targets = this.props.targetUsersForUpload;
+    var sorted = _.sortBy(targets, function(targetId) {
+      var targetInfo = allUsers[targetId];
+      if (_.get(targetInfo, ['patient','isOtherPerson'])) {
+        return targetInfo.patient.fullName;
+      }
+      return targetInfo.fullName;
+    });
+
+    var selectorOpts = _.map(sorted, function(targetId) {
+      var targetInfo = allUsers[targetId];
+      var mrn = _.get(targetInfo, ['patient', 'mrn'], '') ?  ' ' +_.get(targetInfo, ['patient', 'mrn']) : '';
+      if (_.get(targetInfo, ['patient','isOtherPerson'])) {
+        return {value: targetId, label: targetInfo.patient.fullName};
```

Comment:
MRN is optional, so any account may not have an MRN. We do expect clinics to be interacting with the old fake child accounts until we come up with and implement the migration from the old child accounts to custodial account relationships where it's relevant. Even then, the user might take ownership of their account and it will be much like the current fake child accounts (if I understand that correctly).